controllers/gateway/ratelimit/ratelimit_controller.go

```diff
@@ -178,3 +204,15 @@ func NewRateLimitReconciler(mgr manager.Manager) *RateLimitReconciler {
 		ReconcilePeriod: defaultReconciliationPeriod,
 	}
 }
+
+func getLatestAPIGatewayCR(apigatewayCRs *operatorv1alpha1.APIGatewayList) *operatorv1alpha1.APIGateway {
```

Comment:
This does not really make sense to be timestamp based. The only APIGateway we care about is the one named `default`